File: examples/developer-hub-javascript/get_feed_id.js
function getFeedId(category, feedName) {
  const hexFeedName = Array.from(feedName)
    .map((c) => c.charCodeAt(0).toString(16).padStart(2, "0"))
    .join("");
  const paddedHexString = (category + hexFeedName).padEnd(42, "0");
  return `0x${paddedHexString}`;
}

const feedId = getFeedId("01", "FLR/USD");
console.log(feedId);

File: examples/developer-hub-rust/src/bin/get_feed_id.rs
fn get_feed_id(category: &str, feed_name: &str) -> String {
    let hex_feed_name = feed_name
        .as_bytes()
        .iter()
        .map(|byte| format!("{byte:02x}"))
        .collect::<Vec<String>>()
        .join("");
    let combined = format!("{category}{hex_feed_name}");
    let padded_hex_string = format!("{combined:0<42}");
    format!("0x{padded_hex_string}")
}

fn main() {
    let feed_id = get_feed_id("01", "FLR/USD");
    println!("{feed_id}");
}
